src/UI/App.jsx

```diff
@@ -304,6 +306,11 @@ export default class App extends React.Component {
 			this.Api.setCustomHeaders(nextState.apiCustomHeaders);
 		}
 
+		if(nextState.messengerOpenState !== this.state.messengerOpenState) {
```

Comment:
Enige wat ik me hierop kan bedenken is waarom staat dit in `ShouldComponentUpdate` en niet in `getDerivedStateFromProps`.
Enige wat should update zou moeten doen is bepalen of het wel of niet zou moeten updaten.
https://legacy.reactjs.org/docs/react-component.html
![image](https://github.com/parley-messaging/web-library/assets/49270238/e505c803-c589-4e35-afca-bf9ae8e700c3)

Reply:
^ Al is dit wel een beetje out of scope sorry :P

Reply:
Nou opzich goeie vraag.
Antwoord voor deze specifieke change is; al dit soort updates staan hierin, dus daarom deze ook

Antwoord op de vraag waarom uberhaupt state change checks in `shouldComponentUpdate()` en niet `getDerivedStateFromProps()`; 

volgens mij heeft `getDerivedStateFromProps()` geen toegang tot de huidige state (`this.state`) aangezien die `static` is en 
> This method doesn’t have access to the component instance.

source: https://legacy.reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Dus dan blijft over `shouldComponentUpdate()` en `componentDidUpdate()` en daar heb ik niet echt een antwoord op..

Reply:
Misschien moeten we hier een keer samen voor zitten (met Daan?) om hier in te duiken. Ik dacht inderdaad dat `getDerivedStateFromProps` ook prevState en prevProps had maar dat lijkt hij niet te hebben als ik naar de docs kijk.
Maar het hoort eigenlijk niet thuis in shouldComponentUpdate om meerdere redenen, dus laten we samen even kijken waar die eigenlijk wel word.